### PR TITLE
New Pull Request

### PR DESCRIPTION
**Pull request by Chenxiao-Tian**

### CODE REVIEW

**Review by Chenxiao-Tian — COMMENTED** [review]
Fix Some Typos

**Comment by jyl17** [comment]
I guess you want to merge the commits into main rather than other branches

File: tests/test_sampling.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+def test_direct_sampling_with_times():
+    Q = np.array([[-1, 0.5, 0.5],
+                  [0.3, -0.8, 0.5],
+                  [0.4, 0.2, -0.6]])
+    T = 13
+    result = direct_sampling_with_times(Q, T)
+    
+    # Test for output type
+    assert isinstance(result, list), "Output should be a list"
```
Comment (Joejoe12341234) [inline tests/test_sampling.py:13]: ## If the result list is supposed to have a specific length, test for that
    expected_length = [the expected length]
    assert len(result) == expected_length, "The output list should have a length of expected_length"

File: tests/test_sampling.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import pytest
+from your_script_name import direct_sampling_with_times  # Replace with your actual script name
+import numpy as np
+
```
Comment (Joejoe12341234) [inline tests/test_sampling.py:4]: ## Test for Performance Constraints

import time

def test_performance():
    start_time = time.time()
    direct_sampling_with_times(Q, T)
    end_time = time.time()
    assert end_time - start_time < max_allowed_time

File: tests/ci.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.8'  # Or any version you need
```
Comment (Joejoe12341234) [inline tests/ci.yml:15]: # Consider testing Across Multiple Python versions
 python-version: [3.7, 3.8, 3.9, 3.10]  # List all supported versions

File: Direct Sample.py [hunk 4]
```diff
@@ -51,7 +51,7 @@ def direct_sampling_with_times(Q, T):
               [0.4, 0.2, -0.6]])
 
 # Define a time interval T
-T = 10  # arbitrary time interval
+T = 13  # arbitrary time interval
```
Comment (Joejoe12341234) [inline Direct Sample.py:54]: Consider changing variable name: time_interval = 13

**Pull request merged**